hail/python/hail/vds/combiner/variant_dataset_combiner.py

```diff
@@ -248,29 +249,27 @@ def _step_gvcfs(self):
                                                 self.gvcf_import_intervals,
                                                 array_elements_required=False,
                                                 _external_header=self.gvcf_external_header,
-                                                _external_sample_ids=[[name] for name in names] if sample_names is not None else None,
+                                                _external_sample_ids=[[name] for name in names] if names is not None else None,
                                                 reference_genome=self.reference_genome,
                                                 contig_recoding=self.contig_recoding)]
             merge_vds.append(combine_variant_datasets(vdses))
+            merge_n_samples.append(len(vdses))
         if self.finished and len(merge_vds) == 1:
             merge_vds[0].write(self.output_path)
             return
 
         temp_path = self._temp_out_path(f'gvcf-combine_job{self._job_id}/dataset_')
         pad = len(str(len(merge_vds) - 1))
-        merge_metadata = []
-        count = 0
-        for _ in merge_vds:
-            merge_metadata.append(VDSMetadata(path=temp_path + str(count).rjust(pad, '0') + '.vds',
-                                  n_samples=len(vdses)))
-            count += 1
```

Comment:
this was wrong, `len(vdses)` is not in general the number of samples of all datasets in `merge_vds`.